### PR TITLE
[RFR] [ClassContent] #213 updated ::jsonSerialize and added ::getDefaultImageName and associated tests

### PR DESCRIPTION
**Pull request by eric-chau**
Added ``AClassContent::getDefaultImageName`` and associated tests. Updated ``AClassContent::jsonSerialize`` and added tests for ``AClassContent::jsonSerialize`` and ``AContent::jsonSerialize``.

### CODE REVIEW

File: ClassContent/AClassContent.php [hunk 1]
```diff
@@ -853,7 +853,12 @@ public function getMode()
      */
     public function getImageName()
     {
-        return str_replace(array(self::CLASSCONTENT_BASE_NAMESPACE, '\\'), array('', '/'), get_class($this)).'.png';
+        return $this->getDefaultImageName();
+    }
+
+    public function getDefaultImageName()
```
Comment (mickaelandrieu) [inline ClassContent/AClassContent.php]: why ``getImageName()`` return ``getDefaultImageName()`` ?

File: ClassContent/AClassContent.php [hunk 2]
```diff
@@ -1091,7 +1098,10 @@ public function jsonSerialize($format = self::JSON_DEFAULT_FORMAT)
 
         $data = array_merge([
             'properties' => $this->getProperty(),
-            'image'      => $this->getImageName(),
+            'image'      => self::JSON_DEFINITION_FORMAT === $format
+                ? $this->getDefaultImageName()
+                : $this->getImageName()
```
Comment (mickaelandrieu) [inline ClassContent/AClassContent.php]: in all cases, you return ``getImageName()`` (cf previous comment)

File: ClassContent/Tests/ClassContentTest.php [hunk 3]
```diff
@@ -221,4 +224,108 @@ public function prepareCommitDraft()
...
         }
     }
+
+    public function testJsonSerialize()
+    {
+        $this->assertInstanceOf('JsonSerializable', $this->content);
+
+        $data = $this->content->JsonSerialize();
+
+        $this->assertTrue(is_array($data));
```
Comment (mickaelandrieu) [inline ClassContent/Tests/ClassContentTest.php]: You should use a DataProvider :)

**Comment by mickaelandrieu** [comment]
:+1:

File: ClassContent/AClassContent.php (hunk 1 above)
Comment (eric-chau) [inline ClassContent/AClassContent.php]: You're right I didnt add docblock to explain this part, I will, thanks.

File: ClassContent/AClassContent.php (hunk 2 above)
Comment (eric-chau) [inline ClassContent/AClassContent.php]: You're right but when I will do my change, you will be able to override ``getImageName()`` but not ``getDefaultImageName()``.

File: ClassContent/Tests/ClassContentTest.php (hunk 3 above)
Comment (eric-chau) [inline ClassContent/Tests/ClassContentTest.php]: I don't get what you want to mean :o

Comment (mickaelandrieu) [inline ClassContent/Tests/ClassContentTest.php]: You can use @DataProvider annotation avoid multiples assertions. See PHPUnit docs, it's easy to use and then your tests are easier to maintain  :)

Comment (eric-chau) [inline ClassContent/Tests/ClassContentTest.php]: Ok thanks but this doesn't suit what I aim to do.

File: ClassContent/AClassContent.php (hunk 2 above)
Comment (eric-chau) [inline ClassContent/AClassContent.php]: Ping @mickaelandrieu, I updated ``::getDefaultImageName`` to be a final method, let me know if it's clearer for you.

Comment (mickaelandrieu) [inline ClassContent/AClassContent.php]: :+1:

**Comment by mickaelandrieu** [comment]
go for merge, good job !

**Pull request merged**